org.eclipse.m2e.pde.connector/src/org/eclipse/m2e/pde/connector/TychoPackagingsConfigurator.java

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Konrad Windszus
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *      Konrad Windszus
+ *******************************************************************************/
+package org.eclipse.m2e.pde.connector;
+
+import java.util.List;
+
+import org.apache.maven.plugin.MojoExecution;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.ProjectScope;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.m2e.core.internal.markers.SourceLocationHelper;
+import org.eclipse.m2e.core.project.configurator.AbstractProjectConfigurator;
+import org.eclipse.m2e.core.project.configurator.ProjectConfigurationRequest;
+import org.eclipse.pde.ds.internal.annotations.DSAnnotationVersion;
+import org.eclipse.pde.internal.core.natures.PDE;
+import org.osgi.framework.Version;
+
+@SuppressWarnings("restriction")
+public class TychoPackagingsConfigurator extends AbstractProjectConfigurator {
+
+	private static final String TYCHO_GROUP_ID = "org.eclipse.tycho";
+	private static final String TYCHO_DS_PLUGIN_ARTIFACT_ID = "tycho-ds-plugin";
+	private static final String GOAL_DECLARATIVE_SERVICES = "declarative-services";
+
+	@Override
+	public void configure(ProjectConfigurationRequest request, IProgressMonitor monitor) throws CoreException {
+		MavenProject mavenProject = request.mavenProject();
+		IProject project = request.mavenProjectFacade().getProject();
+		String packaging = mavenProject.getPackaging();
+		if ("eclipse-plugin".equals(packaging) || "eclipse-test-plugin".equals(packaging)) {
```

Comment:
probably there should also test class folders be added to the Java Build Path (compare with https://github.com/eclipse-tycho/tycho/issues/1491). Is there any recommended code layout for unit tests inside `eclipse-plugin`?

Reply:
> Is there any recommended code layout for unit tests inside `eclipse-plugin`?

You mean in general or for testing this connector?
At the moment I don't think there is a general recommendation since Plug-in projects cannot (yet) specify test-dependencies and the usual pattern is to separate tests into fragments.
However I would make it similar to the maven structure and have a separate `src-test` or `test-src` folder where the sources are placed.

> probably there should also test class folders be added to the Java Build Path (compare with [eclipse-tycho/tycho#1491](https://github.com/eclipse-tycho/tycho/issues/1491)). 

For a `eclipse-test-plugin` that would be reasonable, yes.
Do you want to include that into this PR or create a follow-up?

Reply:
`eclipse-plugin-test` is nothing we should specifically support as it is actually nothing special and has only a meaning when executing Tycho. Even there I always would recommend to use `eclipse-test` instead.

Whether or not a folder is meant as a test-folder can not be derived from the Tycho configuration, you can *guess* it from the standard maven layout, but m2e already does that.

Reply:
> `eclipse-plugin-test` is nothing we should specifically support as it is actually nothing special and has only a meaning when executing Tycho. Even there I always would recommend to use `eclipse-test` instead.
> 

Is  `eclipse-test` a new packaging type? I don't know about that.

However since that topic seems not yet settled, lets merge this PR now and improve/enhance in with follow up PRs.

Reply:
> Is eclipse-test a new packaging type? I don't know about that.

Should be `eclipse-plugin` with `test` folder ;-)